### ISSUE TITLE
Individual fast and slow parameters (from the python interpreter)

### ISSUE THREAD

**Issue by joezuntz** [CobayaSampler/cobaya#1]
Hi there,

Thanks for this great code!

I was looking and couldn't immediately see a way to set individual parameters to be fast and slow when using the python interface.  e.g. if I have some generic likelihood function:

```
def my_logp(x,y):
    # function with some clever caching so that changing y is fast
    return logp

# Copied from your example
info = {"likelihood": {"my_like": my_logp}}
info["params"] = odict([
    ["x", {"prior": {"min": -2, "max": 2}, "ref": 1, "proposal": 0.2}],
    ["y", {"prior": {"min": -2, "max": 2}, "ref": 0, "proposal": 0.2}]])
```

Is there some way to do this for the individual parameters as opposed to the whole likelihood?

Cheers,
Joe

```

**Comment by JesusTorrado** [comment]
Hi Joe,

Thanks for the first issue! Indeed, there is no way right now to specify speeds per parameter. We'll look into it for a future 1.X release.

As a temporary workaround, if you mention the faster parameter info *last* (must be using an OrderedDict), it will be varied independently (won't be mixed with the rest of the likelihood parameters).

Cheers
Jesús

**Comment by joezuntz** [comment]
Hi Jesus,

Many thanks for this!  I'll try it out.

**Comment by joezuntz** [comment]
(sorry, clicked "close" button by mistake)

**Comment by vivianmiranda** [comment]
I was able to go around this by creating "fake" likelihood that just returns zero but "needs" some of the DES parameters that I want to be slower (in comparison to halo bias and multiplicative bias which are insanely fast) - and then I set the speed parameters of these fake likelihoods to be 0. It worked.

**Comment by JesusTorrado** [comment]
That's great! Let this be the official workaround for now.

**Comment by JesusTorrado** [comment]
Code to solve this is on the way. Stay tuned!

**Comment by vivianmiranda** [comment]
You are the best :)

Vivian Miranda

On Mar 28, 2019, at 2:23 PM, Jesús Torrado <notifications@github.com<mailto:notifications@github.com>> wrote:


Code to solve this is on the way. Stay tuned!

**Comment by JesusTorrado** [comment]
Fixed in 1.1 (coming tomorrow). Check out https://cobaya.readthedocs.io/en/latest/sampler_mcmc.html#taking-advantage-of-a-speed-hierarchy
